spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/spanner/ReplacingGrpcCallInterceptor.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2021-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.autoconfigure.spanner;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.Value;
+import com.google.spanner.v1.BatchCreateSessionsResponse;
+import com.google.spanner.v1.CommitResponse;
+import com.google.spanner.v1.PartialResultSet;
+import com.google.spanner.v1.ResultSetMetadata;
+import com.google.spanner.v1.Session;
+import com.google.spanner.v1.StructType;
+import com.google.spanner.v1.Transaction;
+import com.google.spanner.v1.Type;
+import com.google.spanner.v1.TypeCode;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+
+/**
+ * gRPC interceptor that stops the chain and returns a pre-set response object.
```

Comment:
So it's basically a Mock? If so, have you looked at https://github.com/Fadelis/grpcmock ?

Reply:
It is, an I have not seen the `grpcmock` project before. It looks good! For spanner, we could use it by making the test think it's running against an emulator on localhost while actually using the mocked endpoint.

On the other hand, I have two concerns: 1) `grpcmock` starts a grpc server, which is more heavyweight than intercepting all requests in the test, and 2) this is bringing another dependency, and a single-person project for <50 lines of code. The irritating part of mocking is creating the response objects, which we still have to do (life is too short to construct a `PartialResultSet`, but here we are...)

So I could go either way.

Reply:
I definitely agree on the risks of introducing a dependency than may become abandoned, especially given how fast Spring Boot moves and there may be an eventual incompatibility. 

That being said I don't know how many other tests you plan to write against this intercepter, and if lets say its 2+, I'd suggest using the library. Otherwise lets just keep it simple and use your mock/fake.